posthog/temporal/weekly_digest/activities.py

```diff
@@ -167,7 +171,8 @@ async def generate_user_notification_lookup(input: GenerateDigestDataInput) -> N
         user_count = 0
 
         async with redis.from_url(_redis_url(input.common)) as r:
-            async for team in query_teams_for_digest():
+            batch_start, batch_end = input.batch
+            async for team in query_teams_for_digest()[batch_start:batch_end]:
```

Comment:
here we're loading all teams from SQL to just throw away those outside the slice

we could pass the batch start and end through and use them as slice on the queryset and we'll only load that slice from SQL

probably not a big deal here but imho a good habit to get into

Reply:
DOH!

yep, that's the same 🫠